registrator.go

```diff
@@ -172,6 +173,10 @@ func main() {
 			go b.Add(msg.ID)
 		case "die":
 			go b.RemoveOnExit(msg.ID)
+		case "kill":
+			if *deregisterOnStop {
+				go b.RemoveOnExit(msg.ID)
+			}
 		}
```

Comment:
Alternatively, we could further restrict deregistration in this case by checking whether the signal was 15, and only deregistering then. Thoughts?